GW2EIEvtcParser/EIData/Actors/AbstractSingleActor.cs

```diff
@@ -1,1143 +1,915 @@
         {
-            // Trim
-            DespawnEvent despawnCheck = log.CombatData.GetDespawnEvents(agentItem).LastOrDefault();
-            SpawnEvent spawnCheck = log.CombatData.GetSpawnEvents(agentItem).LastOrDefault();
-            DeadEvent deathCheck = log.CombatData.GetDeadEvents(agentItem).LastOrDefault();
-            AliveEvent aliveCheck = log.CombatData.GetAliveEvents(agentItem).LastOrDefault();
-            if (deathCheck != null && (aliveCheck == null || aliveCheck.Time < deathCheck.Time))
-            {
-                replay.Trim(agentItem.FirstAware, deathCheck.Time);
-            }
-            else if (despawnCheck != null && (spawnCheck == null || spawnCheck.Time < despawnCheck.Time))
+            long denom = next.Value.Time - prev.Value.Time;
+            if (denom == 0)
             {
-                replay.Trim(agentItem.FirstAware, despawnCheck.Time);
+                position = prev.Value.Value;
             }
             else
             {
-                replay.Trim(agentItem.FirstAware, agentItem.LastAware);
+                float ratio = (float)(time - prev.Value.Time) / denom;
+                position = Vector3.Lerp(prev.Value.Value, next.Value.Value, ratio);
             }
+            return true;
         }
-
-        protected virtual void TrimCombatReplay(ParsedEvtcLog log)
+        else
         {
-
-        }
-
-        protected void InitCombatReplay(ParsedEvtcLog log)
-        {
-            CombatReplay = new CombatReplay(log);
-            if (!log.CombatData.HasMovementData)
+            var parametric = prev ?? next;
+            if(parametric != null)
             {
-                // no combat replay support on fight
-                return;
+                position = parametric.Value.Value;
+                return true;
             }
-            SetMovements(log);
-            CombatReplay.PollingRate(log.FightData.FightDuration, AgentItem.Type == AgentItem.AgentType.Player);
-            TrimCombatReplay(log);
-            if (!IsFakeActor)
+            else
             {
-                InitAdditionalCombatReplayData(log);
+                position = default;
+                return false;
             }
         }
+    }
 
-        internal IReadOnlyList<GenericDecorationRenderingDescription> GetCombatReplayDecorationRenderableDescriptions(CombatReplayMap map, ParsedEvtcLog log, Dictionary<long, SkillItem> usedSkills, Dictionary<long, Buff> usedBuffs)
-        {
-
-            if (CombatReplay == null)
-            {
-                InitCombatReplay(log);
-            }
-            return CombatReplay.Decorations.GetCombatReplayRenderableDescriptions(map, log, usedSkills, usedBuffs);
-        }
-        protected virtual void InitAdditionalCombatReplayData(ParsedEvtcLog log)
+    //TODO(Rennorb) @cleanup: There is an argument to be made here that in all the places where this is used you probably want to add a decoration either way, regardless if this fails or not.
```

Comment:
Could you explain what do you mean by this? Places that use TryGetCurrentFacingDirection are making use of the vector to give a proper orientation to the decoration.

Reply:
So this functions gives you the facing direction of its actor - but it can fail. 
So what do we do when this fails? Iirc in most cases we don't do the thing we originally wanted to do, so for example we don't create a decoration. but this is wired; my argument is that we probably want to create that decoration either way, and maybe just give it the wrong facing direction. Just because we don't have a direction doesn't mean we don't have a position.

The todo is on the getter because this is the centralized place from which you can find all the usages. It was easier than investigating every single one when i thought about that potential "problem".

Reply:
Why would we want to show a decoration that is facing the wrong way though? ^^ A rectangle or a pie facing the wrong way completely changes the meaning of the mechanic. I don't believe we use Facing directions for decorations that are invariant by rotation.

Reply:
Yea i also figured that exact issues with my original proposal, so maybe this is not the best solution. But not showing anything at all is even worse imo. Thats the reason why there is a not about wanting to look into this more.